include/boost/beast/core/detail/buffer_utils.hpp

```diff
@@ -0,0 +1,57 @@
+//
+// Copyright (c) 2020 Richard Hodges (hodges.r@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+#ifndef BOOST_BEAST_CORE_DETAIL_BUFFER_UTILS_HPP
+#define BOOST_BEAST_CORE_DETAIL_BUFFER_UTILS_HPP
+
+#include <boost/beast/core/detail/config.hpp>
+#include <boost/asio/buffer.hpp>
+#include <algorithm>
+#include <utility>
+#include <type_traits>
+
+namespace boost {
+namespace beast {
+namespace detail {
+
+template<class T>
+struct is_net_buffer
+    : std::false_type
+{
+};
+
+template<>
+struct is_net_buffer<net::mutable_buffer>
+    : std::true_type
+{
+};
+template<>
+struct is_net_buffer<net::const_buffer>
+    : std::true_type
+{
+};
+
+template<class Buffer,
+    typename std::enable_if<is_net_buffer<Buffer>::value>::type * = nullptr>
+auto
+truncate(
+    Buffer &target,
+    std::size_t max_len)
+-> Buffer &
+{
+    target = Buffer(target.data(),
+                    (std::min)(max_len, target.size()));
+    return target;
+}
```

Comment:
This goes in `boost/beast/core/detail/buffer.hpp` and it should be two inline function overloads, no templates